tokio/src/sync/semaphore.rs

```diff
@@ -142,6 +142,13 @@ impl Semaphore {
             Err(_) => Err(TryAcquireError(())),
         }
     }
+
+    /// Closes the semaphore.
+    ///
+    /// This prevents the semaphore from issuing new permits and notifies all pending waiters.
+    pub fn close(&self) {
```

Comment:
What happens if you call `acquire` after closing it? I see that `acquire` contains an unwrap.

Reply:
Looks like @Darksonn's comment has now been addressed.